codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSHttpProtocolServiceClient.kt

```diff
@@ -67,7 +67,7 @@ class AWSHttpProtocolServiceClient(
                     ConfigProperty(
```

Comment:
In initializers that take in arguments for `awsRetryMode` and `maxAttempts`, default value for retry strategy options should be defaults provider function call using those arguments. 

In initializer that doesn't take in arguments for `awsRetryMode` and `maxAttempts` but needs to set retry strategy options value to something that isn't nil (e.g., the region initializer), we can't pass in any values to defaults provider method call.